pkg/api/v1/atlasbackupschedule_types.go

```diff
@@ -96,6 +99,51 @@ type AtlasBackupSchedule struct {
 	Status status.BackupScheduleStatus `json:"status,omitempty"`
 }
 
+func (in *AtlasBackupSchedule) ToAtlas(clusterID, clusterName string, policy *AtlasBackupPolicy) *mongodbatlas.CloudProviderSnapshotBackupPolicy {
+	atlasPolicy := mongodbatlas.Policy{}
+
+	for _, bpItem := range policy.Spec.Items {
+		atlasPolicy.PolicyItems = append(atlasPolicy.PolicyItems, mongodbatlas.PolicyItem{
+			FrequencyInterval: bpItem.FrequencyInterval,
+			FrequencyType:     strings.ToLower(bpItem.FrequencyType),
+			RetentionValue:    bpItem.RetentionValue,
+			RetentionUnit:     strings.ToLower(bpItem.RetentionUnit),
+		})
+	}
+
+	result := &mongodbatlas.CloudProviderSnapshotBackupPolicy{
+		ClusterName:                       clusterName,
+		ReferenceHourOfDay:                &in.Spec.ReferenceHourOfDay,
+		ReferenceMinuteOfHour:             &in.Spec.ReferenceMinuteOfHour,
+		RestoreWindowDays:                 &in.Spec.RestoreWindowDays,
+		UpdateSnapshots:                   &in.Spec.UpdateSnapshots,
+		Policies:                          []mongodbatlas.Policy{atlasPolicy},
+		AutoExportEnabled:                 &in.Spec.AutoExportEnabled,
+		UseOrgAndGroupNamesInExportPrefix: &in.Spec.UseOrgAndGroupNamesInExportPrefix,
+		CopySettings:                      make([]mongodbatlas.CopySetting, 0, len(in.Spec.CopySettings)),
+	}
+
+	if in.Spec.Export != nil {
+		result.Export = &mongodbatlas.Export{
+			ExportBucketID: in.Spec.Export.ExportBucketID,
+			FrequencyType:  in.Spec.Export.FrequencyType,
+		}
+	}
+
+	for _, copySetting := range in.Spec.CopySettings {
+		result.CopySettings = append(result.CopySettings, mongodbatlas.CopySetting{
+			CloudProvider:     copySetting.CloudProvider,
+			RegionName:        copySetting.RegionName,
+			ReplicationSpecID: copySetting.ReplicationSpecID,
+			ShouldCopyOplogs:  copySetting.ShouldCopyOplogs,
+			Frequencies:       copySetting.Frequencies,
+		})
+	}
+
+	result.ClusterID = clusterID
```

Comment:
Out of curiosity, Why are you setting the cluster ID here and not in the block 114-124? Also, both ID and name are needed?

Reply:
Good catch! This is a leftover from the previous implementation. Fixed